### PR TITLE
Use individual Dart Language Versions for each channel

### PR DESCRIPTION
**Pull request by srawlins**
fixes: https://github.com/dart-lang/dart-pad/issues/2074

Add a Dart Language Version field to the flutter-sdk-version.yaml format.

Bump the language version in beta and dev to 2.15.0-0, which enables constructor-tearoffs.

Flip common_server_api_test.dart to run everything null safe.

### CODE REVIEW

File: flutter-sdk-version.yaml [hunk 1]
```diff
@@ -6,7 +6,15 @@
...
-  stable_version: 2.5.3
-  beta_version: 2.7.0-3.0.pre
-  dev_version: 2.6.0-11.0.pre
-  old_version: 2.5.1
+  stable:
+    flutter_version: 2.5.3
+    dart_language_version: 2.14.0
+  beta:
+    flutter_version: 2.7.0-3.0.pre
+    dart_language_version: 2.15.0
```
Comment (domesticmouse) [inline flutter-sdk-version.yaml:14]: this should probably be `2.15.0-0`

File: flutter-sdk-version.yaml [hunk 2]
```diff
@@ -6,7 +6,15 @@
...
+  dev:
+    flutter_version: 2.6.0-11.0.pre
+    dart_language_version: 2.15.0
```
Comment (domesticmouse) [inline flutter-sdk-version.yaml:17]: likewise

File: flutter-sdk-version.yaml [hunk 3]
```diff
@@ -6,7 +6,15 @@
...
+  old:
+    flutter_version: 2.5.1
+    dart_language_version: 2.14.0
```
Comment (domesticmouse) [inline flutter-sdk-version.yaml:20]: If we make this `2.13.0` we could downgrade `flutter_version` to `2.2.5`. Maybe.

**Review by domesticmouse — APPROVED** [review]

**Comment by domesticmouse** [comment]
Looks like you need to update the tests to play nicely with the different generated `pubspec.yaml`s

File: flutter-sdk-version.yaml (hunk 1 above)
Comment (srawlins) [reply flutter-sdk-version.yaml]: Thanks! Done.

File: flutter-sdk-version.yaml (hunk 2 above)
Comment (srawlins) [reply flutter-sdk-version.yaml]: Done.

File: flutter-sdk-version.yaml (hunk 3 above)
Comment (srawlins) [reply flutter-sdk-version.yaml]: Done.

**Pull request merged**